### ISSUE TITLE
Constructor inheritance

### ISSUE THREAD

**Issue by randombyte-developer** [Kotlin/dukat#248]
I am running dukat version 0.0.28.

```
export class A {
  constructor(p1: string) { }
}

export class B extends A { }
```

produces 

```
external open class A(p1: String)

external open class B : A
```

Is this correct? I think the class `B` should also have the constructor from `A`.

I just tried to start with Kotlin JS and use a JS library in Kotlin, but noticed that the type hints were wrong.

**Comment by Schahen** [comment]
Hi! I'm not sure, let's discuss.
Strictly speaking your code is out of scope of current kotlin functionality since it's not a declaration - as of now we support only declaration conversation. 

So, let's modifiy slightly the example and consider this snippet of code:
```typescript 

declare class A {
  constructor(p1: string)
}

declare class B extends A { }
```

This is translated to (I'm omitting imports and annotations):
```kotlin
external open class A(p1: String)

external open class B : A
```

Which is perfectly valid in that sense that it will compile. 
However! In typescript we can use A and B declarations following way:

```typescript
function main() {
        let a = new A("HEY! I AM A");
        let b = new B("HI! I AM B");
}      
```

while in Kotlin this won't compile:
```kotlin


fun main() {
  val a = A("AAA")
  val b = B()
}
```

So this do feels like a problem to me.  Let me think about it for a while and double-check typescript specs. 

If it is indeed a problem, it's definitely a major one - currently we check compilation errors but runtime errors will fly under our radars.

**Comment by randombyte-developer** [comment]
Sorry for the late reply. 

The defintion file I wanted to use is

**Comment by Schahen** [comment]
Will be released in 0.5.8 now can be checked with dukat@next
